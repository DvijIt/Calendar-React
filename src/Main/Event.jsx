import React from 'react';
import PropTypes from 'prop-types';
import DeleteEventPopup from '../Modal/DeleteEventPopup';

class Event extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      show: false,
    };
  }

  showDeletePopup = () => {
    if (this) {
      this.setState({ show: true });
    }
  }

  closeDeletePopup = () => {
    if (this) {
      this.setState({ show: false });
    }
  }

  handleGetEventData = (e) => {
    const { getEventData } = this.props;
    getEventData(e, this.props);
  }

  render() {
    const {
      id, title, timeStart, timeEnd, description, color, checkDelete,
    } = this.props;

    const myStyle = {
      top: `${timeStart.slice(3)}px`,
      backgroundColor: color || '#6495ed',
    };
    const { show } = this.state;

    return (
      <div
        role="button"
        className="event"
        data-option="event"
        data-id={id}
        style={myStyle}
        onPointerEnter={this.showDeletePopup}
        onPointerLeave={this.closeDeletePopup}
        onClick={this.handleGetEventData}
        onKeyPress={this.handleGetEventData}
        tabIndex={0}
      >
        <h4 className="event-title">{title}</h4>
        <div className="event-time">{`${timeStart} - ${timeEnd}`}</div>
        <div className="event-description">{description}</div>
        {show && <DeleteEventPopup id={id} checkDelete={checkDelete} />}
      </div>
    );
  }
}

Event.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  timeStart: PropTypes.string,
  timeEnd: PropTypes.string,
  description: PropTypes.string,
  color: PropTypes.string,
  checkDelete: PropTypes.func,
  getEventData: PropTypes.func,
};

Event.defaultProps = {
  id: PropTypes.string,
  title: PropTypes.string,
  timeStart: PropTypes.string,
  timeEnd: PropTypes.string,
  description: PropTypes.string,
  color: PropTypes.string,
  checkDelete: PropTypes.func,
  getEventData: PropTypes.func,
};

export default Event;
