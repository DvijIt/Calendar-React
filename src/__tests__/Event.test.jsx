import React from 'react'
import { shallow } from 'enzyme'
import Event from '../Main/Event'

describe('<Event />', () => {
  it('should display Calendar', () => {
    const props = {
      id: '1',
      title: 'title 1',
      timeStart: '20:00',
      timeEnd: '21:00',
      description: 'descr',
      color: '#fff',
      checkDelete: jest.fn()
    }
    const wrappedComponent = shallow(<Event {...props} />)
    expect(wrappedComponent).toMatchSnapshot()
  })
})
