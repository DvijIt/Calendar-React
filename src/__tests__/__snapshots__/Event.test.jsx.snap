// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> should display Calendar 1`] = `
<div
  className="event"
  data-id="1"
  data-option="event"
  onClick={[Function]}
  onPointerEnter={[Function]}
  onPointerLeave={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "top": "00px",
    }
  }
>
  <h4
    className="event-title"
  >
    title 1
  </h4>
  <div
    className="event-time"
  >
    20:00 - 21:00
  </div>
  <div
    className="event-description"
  >
    descr
  </div>
</div>
`;
